### DISCUSSION TITLE
Is it possbile to eslint-disable a script(type=module)?

### DISCUSSION THREAD

**Discussion by wxiaoguang** [eslint/eslint#15856] [General]
When working with backend MVC projects, sometimes the code could be:

```html
<!-- I want to disable eslint for this script block -->
<script type="module">
    window.myGlobal.var = {{.MyData}};  // here is invalid JS syntax, but it's rendered by backend template engine correctly.
</script>

<!--  eslint works well for this block, and it should be enabled -->
<script type="module">
    console.log(window.myGlobal.var);
</script>
```

When using `eslint-plugin-html`, I can not find a proper way to disable the eslint for the script(type=module) block (I do not want ignore the whole file).

ps: I know a trick that: `<script><!-- /* eslint-disable */ --></script>`, but it doesn't work for script(type=module)

**Comment by aladdin-add** [comment]
does the `eslint-disable-line` help?

**Reply by wxiaoguang** [reply]
> does the `eslint-disable-line` help?

No, it doesn't, it reports error: `error  Parsing error: Unexpected token {`

The root reason is that here the `eslint-plugin-html` must skip the whole block before syntax parse, but `eslint-disable-line` only tells eslint to skip the parsed line, so the syntax error comes first.

What we need here is to make `eslint-plugin-html` skip a whole `<script>` block because code inside it is not valid JS syntax.

For traditional `<script>` we can use `<script><!-- /* eslint-disable */ --></script>`.

But for `<script type=module>`, this trick doesn't work anymore because the module script doesn't accept `<!--` HTML comment.

**Reply by aladdin-add** [reply]
well, I think you'll have to ask the maintainer of `eslint-plugin-html` - eslint does not know whether the js is in a html or not.

**Reply by wxiaoguang** [reply]
Yup, I also realized it soon .... (I was thinking that the parser is done inside eslint, after reading the code, I find that it's related to the plugin). thank you very much.
